pkg/k8shandler/elasticsearch.go

```diff
@@ -0,0 +1,300 @@
+    Uri: "_flush/synced",
+  }
+
+  curlESService(clusterName, namespace, payload)
+
+  return payload.Error, (payload.StatusCode == 200)
+}
+
+// This will curl the ES service and provide the certs required for doing so
+//  it will also return the http and string response
+func curlESService(clusterName, namespace string, payload *esCurlStruct) {
+
+  urlString := fmt.Sprintf("https://%s.%s.svc:9200/%s", clusterName, namespace, payload.Uri)
+  urlURL, err := url.Parse(urlString)
+
+  if err != nil {
+    logrus.Warnf("Unable to parse URL %v: %v", urlString, err)
+    return
+  }
+
+  request := &http.Request{
+    Method: payload.Method,
+    URL: urlURL,
+  }
+
+  switch (payload.Method) {
+  case http.MethodGet:
+    // no more to do to request...
+  case http.MethodPost:
+    if payload.RequestBody != "" {
+      // add to the request
+      request.Header = map[string][]string{
+        "Content-Type": []string{
+          "application/json",
+        },
+      }
+      request.Body = ioutil.NopCloser(bytes.NewReader([]byte(payload.RequestBody)))
+    }
+
+  case http.MethodPut:
+    if payload.RequestBody != "" {
+      // add to the request
+      request.Header = map[string][]string{
+        "Content-Type": []string{
+          "application/json",
+        },
+      }
+      request.Body = ioutil.NopCloser(bytes.NewReader([]byte(payload.RequestBody)))
+    }
+
+  default:
+    // unsupported method -- do nothing
+    return
+  }
+
+  client := getClient(clusterName, namespace)
+  resp, err := client.Do(request)
+
+  if resp != nil {
+    payload.StatusCode = resp.StatusCode
+    payload.ResponseBody = getMapFromBody(resp.Body)
+  }
+  payload.Error = err
+}
+
+func getRootCA(clusterName, namespace string) *x509.CertPool {
+  certPool := x509.NewCertPool()
+
+  // load cert into []byte
+  caPem, err := ioutil.ReadFile(path.Join(certLocalPath, clusterName, "admin-ca"))
+	if err != nil {
+		logrus.Errorf("Unable to read file to get contents: %v", err)
+		return nil
+	}
+
+  certPool.AppendCertsFromPEM(caPem)
+
+  return certPool
+}
+
+func getMapFromBody(body io.ReadCloser) map[string]interface{} {
+  buf := new(bytes.Buffer)
+  buf.ReadFrom(body)
+
+  var results map[string]interface{}
+  err := json.Unmarshal([]byte(buf.String()), &results)
+  if err != nil {
+
+  }
+
+  return results
+}
+
+func getClientCertificates(clusterName, namespace string) []tls.Certificate {
+  certificate, err := tls.LoadX509KeyPair(
+    path.Join(certLocalPath, clusterName, "admin-cert"),
+    path.Join(certLocalPath, clusterName, "admin-key"),
+  )
+  if err != nil {
+    return []tls.Certificate{}
+  }
+
+  return []tls.Certificate{
+    certificate,
+  }
+}
+
+func getClient(clusterName, namespace string) *http.Client {
+
+  // get the contents of the secret
+  extractSecret(clusterName, namespace)
+
+  return &http.Client{
+    Transport: &http.Transport{
+      TLSClientConfig: &tls.Config{
+        InsecureSkipVerify: true,
```

Comment:
currently the certs used for testing were for names `elasticsearch` and fails verification given the service is `example-elasticsearch`